### PR TITLE
[Keyboard] Add MelGeek MJ6xy which is a 60% keyboard that support mul…

### PR DESCRIPTION
**Pull request by mj-bomb**
…ti-layous

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

**Comment by mj-bomb** [comment]
@fauxpark @labria   hello, please kindly review this PR, Thank you very much

File: keyboards/melgeek/mj6xy/rev3/config.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+#define LED_CAPS_LOCK_PIN C7  
+#define LED_PIN_ON_STATE  0
+
+#define BACKLIGHT_PIN B7
+
+#define RGB_DI_PIN    C6
+
+#ifdef  BACKLIGHT_PIN
+#	define BACKLIGHT_LEVELS    10
```
Comment (fauxpark) [inline keyboards/melgeek/mj6xy/rev3/config.h:38]:
```suggestion
#	define BACKLIGHT_LEVELS    10
```

File: keyboards/melgeek/mj6xy/rev3/rules.mk [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# MCU name
+MCU = atmega32u4
+BOOTLOADER = atmel-dfu
+
```
Comment (fauxpark) [inline keyboards/melgeek/mj6xy/rev3/rules.mk:4]:
```suggestion

# Bootloader selection
BOOTLOADER = atmel-dfu
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by mj-bomb** [comment]
@tzarc @zvecr  hello, please kindly review this PR, Thank you very much

**Comment by drashna** [comment]
Thanks

**Pull request merged**